File: backend/src/main/java/com/culinaryheaven/domain/contest/dto/response/ContestResponse.java
package com.culinaryheaven.domain.contest.dto.response;

import com.culinaryheaven.domain.contest.domain.Contest;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;

public record ContestResponse(

        @Schema(description = "대회 ID", example = "1")
        Long id,

        @Schema(description = "대회 설명", example = "창의로운 두부 요리를 뽐내봐요")
        String description,

        @Schema(description = "대회 시작 시간", example = "2024-11-18T00:00:00")
        LocalDateTime startDate,

        @Schema(description = "대회 종료 시간", example = "2024-11-25T23:59:59")
        LocalDateTime endDate,

        @Schema(description = "주재료 정보", example =
                """                        
                        {
                            "id": 1,
                            "name": "두부",
                            "image": "https://image.url.com/tofu"                           
                        }    
                        
                        """)
        TopicIngredientResponse topicIngredient
) {
    public static ContestResponse of(Contest contest) {
        return new ContestResponse(
                contest.getId(),
                contest.getDescription(),
                contest.getStartDate(),
                contest.getEndDate(),
                TopicIngredientResponse.of(contest.getTopicIngredient())
        );
    }
}

File: android/data/src/main/java/com/jae464/data/remote/model/response/ContestResponse.kt
package com.jae464.data.remote.model.response

import com.jae464.data.util.adjustLocalhostUrl
import com.jae464.domain.model.Contest
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class ContestResponse(
    val id: Long,
    val description: String,
    val startDate: String,
    val endDate: String,
    val topicIngredient: TopicIngredientResponse
)

@Serializable
data class TopicIngredientResponse(
    val id: Long,
    val name: String,
    val image: String,
)

fun ContestResponse.toDomain() = Contest(
    id = id,
    description = description,
    startDate = LocalDateTime.parse(startDate),
    ingredient = topicIngredient.name,
    imageUrl = adjustLocalhostUrl(topicIngredient.image)
)


